### PR TITLE
CATROID-570 Refactor DeleteItemOfUserListActionTest to Kotlin

### PR DESCRIPTION
**Pull request by hofcle78**
*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by qlin93 — CHANGES_REQUESTED** [review]
Good job! Please update the year range in the license header to -2021. 
Thanks!

**Comment by hofcle78** [comment]
Done.

**Review by qlin93 — APPROVED** [review]

File: catroid/src/test/java/org/catrobat/catroid/test/content/actions/DeleteItemOfUserListActionTest.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0}")
+        fun parameters() = listOf(
+            arrayOf("DeleteFirstEntry", listOf(Formula(1)), mutableListOf(1, 2, 3), listOf(2, 3)),
+            arrayOf("DeleteMiddleEntry", listOf(Formula(2)), mutableListOf(1, 2, 3), listOf(1, 3)),
+            arrayOf("DeleteMultipleEntries", listOf(Formula(1), Formula(2)), mutableListOf(1, 2, 3), listOf(2)),
+            arrayOf("DeleteAllEntries", listOf(Formula(1), Formula(1)), mutableListOf("First", 2), emptyList<Any>()),
+            arrayOf("DeleteInvalidEntries", listOf(Formula(3), Formula(0), Formula(-1)), mutableListOf(1, 2), listOf(1, 2)),
+            arrayOf("DeleteNotANumberEntry", listOf(Formula(Double.NaN)), mutableListOf(1, 2), listOf(2)),
```
Comment (wslany) [inline catroid/src/test/java/org/catrobat/catroid/test/content/actions/DeleteItemOfUserListActionTest.kt:56]:
```suggestion
            arrayOf("DeleteNotANumberEntry", listOf(Formula(Double.NaN)), mutableListOf(1, 2), listOf(1, 2)),
```
The removal is actually a bug, an incorrect behavior of the implementation. The correct behavior should be that the item is not removed (note: it is correctly implemented in Scratch and in Catty). Please correct the bug (preferred) or remove this line, and also in the CLT, here:

image

**Review by wslany — CHANGES_REQUESTED** [review]
Unfortunately, Catroid has a bug here. Two ways to solve it, see inlined comment.

File: catroid/src/test/java/org/catrobat/catroid/test/content/actions/DeleteItemOfUserListActionTest.kt (hunk 1 above)
Comment (hofcle78) [reply catroid/src/test/java/org/catrobat/catroid/test/content/actions/DeleteItemOfUserListActionTest.kt]: The Bug is now fixed and the CLT is accordingly modified.

**Review by qlin93 — APPROVED** [review]
Looks good

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**